src/GovUk.Education.ExploreEducationStatistics.Admin/Requests/Public.Data/BatchMappingUpdateRequests.cs

```diff
@@ -0,0 +1,72 @@
+#nullable enable
+using System.Collections.Generic;
+using System.Linq;
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Admin.Validators;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Requests.Public.Data;
+
+public record BatchLocationMappingUpdatesRequest
+{
+    public List<LocationMappingUpdateRequest> Updates { get; init; } = [];
+
+    public class Validator : AbstractValidator<BatchLocationMappingUpdatesRequest>
+    {
+        public Validator()
+        {
+            RuleForEach(request => request.Updates)
+                .SetValidator(new LocationMappingUpdateRequest.Validator());
+        }
+    }
+}
+
+public record LocationMappingUpdateRequest
+{
+    public GeographicLevel? Level { get; init; }
+
+    public string SourceKey { get; init; } = string.Empty;
+
+    public string? CandidateKey { get; init; }
+
+    public MappingType? Type { get; init; }
+
+    public class Validator : AbstractValidator<LocationMappingUpdateRequest>
+    {
+        private static readonly MappingType[] ManualMappingTypes =
+        [
+            MappingType.ManualNone,
+            MappingType.ManualMapped
+        ];
+
+        public Validator()
+        {
+            RuleFor(request => request.Level)
+                .NotNull()
+                .IsInEnum();
+
+            RuleFor(request => request.SourceKey)
+                .NotEmpty();
+
+            RuleFor(request => request.Type)
+                .Cascade(CascadeMode.Stop)
+                .NotNull()
+                .Must(type => ManualMappingTypes.Contains(type!.Value))
+                .WithErrorCode(ValidationMessages.ManualMappingTypeInvalid.Code)
+                .WithMessage(ValidationMessages.ManualMappingTypeInvalid.Message);
+
+            RuleFor(request => request.CandidateKey)
+                .NotEmpty()
+                .When(request => request.Type is not null && request.Type.Value == MappingType.ManualMapped)
+                .WithErrorCode(ValidationMessages.CandidateKeyMustBeSpecifiedWithMappedMappingType.Code)
+                .WithMessage(ValidationMessages.CandidateKeyMustBeSpecifiedWithMappedMappingType.Message);
+            
+            RuleFor(request => request.CandidateKey)
+                .Null()
+                .When(request => request.Type is not null && request.Type.Value != MappingType.ManualMapped)
```

Comment:
wonder if this should be `.When(request => request.Type is not null && request.Type.Value == MappingType.ManualNone)` to match the error code more accurately

Reply:
Just seen that you've fixed this in (EES-5115)[https://github.com/dfe-analytical-services/explore-education-statistics/pull/5028] so I'm happy for you to leave this as-is here as it will get fixed when 5115 gets merged :)